controller/resource.go

```diff
@@ -61,42 +51,16 @@ func (c *ResourceController) Read(ctx *app.ReadResourceContext) error {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("not a service account"))
 	}
 
-	var res *resource.Resource
-	var scopes []resourceType.ResourceTypeScope
-
-	err := transaction.Transactional(c.app, func(tr transaction.TransactionalResources) error {
-
-		var error error
-		// Load the resource
-		res, error = tr.ResourceRepository().Load(ctx, ctx.ResourceID)
-
-		if error != nil {
-			return error
-		}
-
-		// Load the resource type scopes
-		scopes, error = tr.ResourceTypeScopeRepository().LookupForType(ctx, res.ResourceTypeID)
-
-		return error
-	})
-
+	svc := resourceservice.NewResourceService(c.app, c.app)
```

Comment:
Add `ResourceService()` method to [Services](https://github.com/fabric8-services/fabric8-auth/blob/master/application/service/services.go) interface, then you can just do:

`svc := c.app.ResourceService()`

(Will also need to add implementation [here](https://github.com/fabric8-services/fabric8-auth/blob/master/gormapplication/application.go))

Reply:
Done in https://github.com/fabric8-services/fabric8-auth/pull/495/commits/45158c56db15e82d8b7f3e29a52175f9a1e34025